pkg/cloud/gke/gcloud.go

```diff
@@ -562,10 +563,31 @@ func StreamTransferFileFromBucket(fullBucketURL string) (*bufio.Scanner, error)
 	if err != nil {
 		return nil, err
 	}
-	err = cmd.Start()
+
+	stderr, err := cmd.StderrPipe()
 	if err != nil {
 		return nil, err
 	}
+
+	if err = cmd.Start(); err != nil {
+		return nil, err
+	}
+
+	if stderr != nil {
+		scanner := bufio.NewScanner(stderr)
+		scanner.Split(bufio.ScanLines)
+		var errorStream bytes.Buffer
+		for scanner.Scan() {
```

Comment:
I don't think this is correct. So without any further doing, I would expect that you call `cmd.Start` which will start the `gsutil` command without waiting. Originally the idea is to then directly return the Scanner and let the caller of the method deal with scanning/processig/streaming the stdout. 

With your change,  I would expect that after `Start` we are not effectively waiting here until EOF in the error stream at which point you break out of this loop, check the stderr and act accordingly. This means effectively that you are back to just synchronously executing `gsutil`. You can have this easier via `cmd.Run`. 

Have you tested this? 

I'd try to split this function into two, a `StreamTransferFileFromBucket` and a `streamCmd` (or something). The idea is to make the streaming (stdout/stderr) handling testable and independent from `gsutil`.

Reply:
If you want to make this a synchronous function, I'd rename `StreamTransferFileFromBucket` to `TransferFileFromBucket` and return the full log as string or[]byte or whatever. Let the caller wrap a `Scanner` around it.

Reply:
@dgozalo - you wrote a lot of this initial logging code. What's your take on this?

Reply:
@romainverduci see also https://github.com/jenkins-x/jx/issues/7027#issuecomment-610522671